### PR TITLE
feat(core): add project description

### LINKED ISSUE #2218: Add `schema:description` to Project metadata
We should add a `schema:description` field to the `Project` metadata to allow moving away from using Gitlabs description field.

This means 
- allowing description to be set when creating a project (through `renku init` and the service)
- adding a `renku project edit` command that allows changing the description (and maybe other fields as well?)
- Adding a `project.edit` endpoint to do the same in the service
- ~Moving some of the other project related endpoints from `cache.*` over to `project.*` in the service. We should keep the old ones around for backwards compatibility and mark them as deprecated.~

### PR DESCRIPTION
**Pull request by mohammad-sdsc**
# Description

Can provide description when initializing a project. If project's template has a `description` attribute, this will be set as its default value and renku won't prompt for that parameter.

/deploy
Fixes #2218

### CODE REVIEW

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
This is a very nice addition! I added a couple of comments.

I would rather not mix this new `description` variable with the variables listed in the template manifest, and pass it to the Jinja templating system as `__description__`. That would require adapting the UI and the templates as well, but it seems a cleaner solution to me.
I'm also fine with keeping the current solution if we prefer to avoid breaking changes.

P.S. There is something that was not mentioned in the issue and is not totally clear to me. If this field should supersede the GitLab description, how should the client (i.e. UI) access this information? Should it come from the KG or should we consider adding a `project.get` endpoint?

**Review by Panaetius — CHANGES_REQUESTED** [review]

**Comment by Panaetius** [comment]
>There is something that was not mentioned in the issue and is not totally clear to me. If this field should supersede the GitLab description, how should the client (i.e. UI) access this information? Should it come from the KG or should we consider adding a project.get endpoint?

It should be in the KG as project metadata ends up there as well

File: renku/service/serializers/project.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Renku service project serializers."""
+from marshmallow import fields
+
+from renku.core.models.dataset import DatasetCreatorsJson as DatasetCreators
+from renku.service.serializers.common import (
+    AsyncSchema,
```
[the remark this answers is not in the document]
Comment (mohammad-sdsc) [reply renku/service/serializers/project.py:23]: All endpoints that write to a repo support async operation including the ones that should run fast like dataset edit or setting a config. I'll keep this for now to be consistent with the rest. We can decide later which endpoints won't benefit from async execution and adjust them.

File: renku/core/commands/init.py [hunk 2]
```diff
@@ -150,14 +150,18 @@ def select_template_from_manifest(
...
-def verify_template_variables(template_data, metadata):
-    """Verifies that template variables are correcly set."""
+def verify_template_variables(template_data, metadata, project_metadata):
+    """Verifies that template variables are correctly set."""
     template_variables = template_data.get("variables", {})
     template_variables_keys = set(template_variables.keys())
+    # NOTE: Set project metadata as template values if there is an exact parameter which has no value
+    for key, value in project_metadata.items():
+        if key in template_variables_keys and key not in metadata:
+            metadata[key] = value
```
[the remark this answers is not in the document]
Comment (mohammad-sdsc) [reply renku/core/commands/init.py]: The idea here was to use project description as the default value for description in the template if template has a description variable. I see your point here and agree that it's cleaner if we don't do such implicit assignment. I remove this feature and pass project's description as `__project_description__` to templates.

**Review by Panaetius — APPROVED** [review]

**Review by lorenzo-cavazzi — APPROVED** [review]
I tested a bit and all works well! Very nice addition! :+1:

**Pull request merged**